python/docs/Makefile

```diff
@@ -16,7 +16,7 @@
 # Minimal makefile for Sphinx documentation
 
 # You can set these variables from the command line.
-SPHINXOPTS    ?= "-W"
+SPHINXOPTS    ?= "-W" "-j" "auto"
```

Comment:
Unfortunately, this breaks Python API docs in many core machines because this means the number of parallel `SparkSubmit` invocation of PySpark. In addition, given that each `PySpark` currently is launched with `local[*]`, this ends up `N * N` `pyspark.daemon`.

In short, this setting is only valid on low-core machine like `GitHub Action` runners (4 cores). This breaks `Python` documentations build even on M3 Max environment and this is worse on large EC2 machines. You can see the failure locally like this.

```
$ build/sbt package -Phive-thriftserver
$ cd python/docs
$ make html
...
java.lang.OutOfMemoryError: Java heap space
...
24/09/16 14:09:55 WARN PythonRunner: Incomplete task 7.0 in stage 30 (TID 177) interrupted: Attempting to kill Python Worker
...
make: *** [html] Error 2
```

Reply:
> As of today, this setting seems to work on low-core machine like `GitHub Action` runners (4 cores).

It seems something about the build has changed since this PR was merged in ~8 months ago. As stated in the PR description, I tested `auto` on a 16-core Intel and it worked fine. And I remember running the build many times to get a representative runtime to report.

> this ends up `N * N` `pyspark.daemon`

Perhaps this was not the case back in January, for some reason. 🤷‍♂️